docs/DeveloperGuide.md

```diff
@@ -1055,22 +1055,176 @@ testers are expected to do more *exploratory* testing.
+    
+### Event management
+
+#### Adding an event
+
+1. Adding an event while all events are being shown
+    1. Test case: `addevent n/Old Event dt/14-04-2020 1600` <br>
+       Expected: No change in the GUI as the event is past event, although if `listevent -a` is run this event should be visable.
+    2. Test case: `addevent n/Future Event dt/14-04-2023 1300` <br>
+       Expected: GUI should be updated with a new event called `Future Event`
+    3. Test case: `addfriend n/Date Check dt/99-04-2023 1240` <br>
+       Expected: Error message should be displayed concerning the incorrect format of date.
+
+2. Adding an event while in another tab
+    1. Prerequisites: Must be in either `Friends` or `Insights` tab.
+
+    2. Test case: `addevent n/Party dt/14-04-2023 d/Cool party` <br>
+       Expected: Tab should switch to `Events` tab and a new event `Party` should be present in the events list.
+
+#### Deleting an event
+
+1. Deleting an event while all events are being shown
+
+    1. Prerequisites: List all events using the `listevents` command. Multiple events in the list.
+
+    2. Test case: `deletevent 1`<br>
+       Expected: First event is deleted from the list. Details of the deleted event shown in the status message.
+
+    3. Test case: `deletefriend 0`<br>
+       Expected: No event is deleted. Error details shown in the status message. Status bar remains the same.
+
+    4. Other incorrect delete commands to try: `deleteevent`, `deleteevent x`, `...` (where x is larger than the list size)<br>
+       Expected: Similar to previous.
+
+#### Editing an event
+
+1. Editing an event while all events are being shown
+
+    1. Prerequisites: Atleast 3 events present, with one their names being `Birthday`
+
+    2. Test case: `editevent 1 n/Alex's Birthday d/Alex's 26th birthday` <br>
+       Expected: Event at index 1 is changed to have the name `Alex's Birthday` and description `Alex's 26th birthday`
+
+    3. Test case: `editevent 1 af/Alex Yeoh rf/Bernice Yu` <br>
+       Expected: `Alex Yeoh` is added to the first event and `Bernice Yu` is removed from the first event.
+
+    4. Test case: `editevent x n/New Event Title` (where x is larger than the list size) <br>
+       Expected: No event edited, index error message provided.
+
+#### Finding an event
+
+1. finding an event while all events are being shown
+    1. Prerequisites: Atleast 3 events present, with one their names being `Birthday`, one of them having a friend `Alex Row` and one of having a starting date of `10-03-2022`
+
+    2. Test case: `findevent n/Birthday` <br>
+       Expected: Events whose names contain the substring `Birthday` are displayed
+
+    3. Test case: `findevent f/Alex Row` <br>
+       Expected: Events whose friends contain `Alex Row` are displayed
+
+    4. Test case: `findevent ds/10-03-2022` <br>
+       Expected: Events which occur after `10-03-2022` are displayed,
+   
+    5. Test case: `findevent n/Birthday f/Alex Row  ds/10-03-2022 1400` <br>
```

Comment:
ds/ takes a date without any time given

Also, would be good to make this clearer by being explicit about how this is an AND check